shredstream.proto

```diff
@@ -0,0 +1,18 @@
+syntax = "proto3";
+
+package shredstream;
+
+import "google/protobuf/empty.proto";
+import "shared.proto";
+
+message Heartbeat {
+  // don't trust IP:PORT from tcp header since it can be tampered over the wire
+  // this also prevents spamming an unwitting destination
+  shared.Socket socket = 1;
+  string auth_token = 2;
```

Comment:
Let's rm this so we're not coupling auth flow with the API def, token will be included as a bearer token in http headers